t/05-scheduler-full.t

```diff
@@ -129,9 +130,17 @@ subtest 're-scheduling and incompletion of jobs when worker is unresponsive or c
 
     # try to allocate to previous worker and fail!
     my ($allocated) = scheduler_step();
+    is(@$allocated, 0, 'no jobs allocated');
 
-    # simulate unresponsive worker which will register itself but not grab any jobs
-    my $unstable_w_pid = unresponsive_worker($k->key, $k->secret, "http://localhost:$mojoport", 3);
+    # simulate a broken worker; it will register itself but declare itself as broken
+    my $broken_w_pid = broken_worker($k->key, $k->secret, "http://localhost:$mojoport", 3, 'out of order');
+    wait_for_worker($schema, 5);
+    $allocated = scheduler_step();
+    is(@$allocated, 0, 'scheduler does not consider broken worker for allocating job');
+    kill_service($broken_w_pid, 1);
+
+    # simulate a rejective worker; it looks free but rejects all jobs assigned to it
+    my $rejective_w_pid = rejective_worker($k->key, $k->secret, "http://localhost:$mojoport", 3, 'rejection reason');
```

Comment:
Before:
 - unresponsive worker, doesn't grab any jobs

After:
- broken worker, declares itself as broken
- rejective worker, rejects all jobs

I'm not clear on the two new ones. What is "broken"?
Also... "rejective" is not a word... did you mean repudiative?

Reply:
Broken means the worker is in the "broken" state. That's actually a well-defined work within the scope of the openQA worker. It is also already used within the UI for the that particular worker status.

Google translator says "rejective" means "ablehnend" and that's exactly what I mean here. And lexico.cpm has a matching definition for it, too. So that worker rejects doing any work (sends the *reject* message introduced by this PR). That is different from the unresponsive worker which does not respond at all (just ignores messages but does not send the *reject* message).

Reply:
Interesting. `lexico.com` shows me the verb or the adjective depending on how fast I type.

I think what's confusing me here is the mix. Say `# Simulate a worker in broken state` would clearly refer to the status string you get. But in the second case it should describe the behavior, like `# Simulate an idle worker that rejects all jobs assigned to it`.
How would that sound to you?